### PR TITLE
Dynamic tab height

### PR DESCRIPTION
**Pull request by John-WL**

### CODE REVIEW

File: javascript/compute-height.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+const getDynamicElementFromContainer = (container) => {
+    const webuiParentDepth = 7;
+    return getRecursiveParentNode(container, webuiParentDepth);
+}
+
+function computeComfyuiElementHeight() {
+    const container = document.getElementById("comfyui_webui_container");
+    const dynamicElement = getDynamicElementFromContainer(container);
+
+    if(dynamicElement !== null) {
```
Comment (ljleb) [inline javascript/compute-height.js:18]: I'd invert the polling condition:

```suggestion
    if(dynamicElement === null) {
        setTimeout(computeComfyuiElementHeight, 200);
        return;
    }
    // rest of function as normal, un-nested code...
```

**Comment by ljleb** [comment]
Is this how it's supposed to look?

image

Except for the footer which is not planned in this PR AFAIK, there seems to be a gap at the top. Do you think we should remove it?

**Comment by ljleb** [comment]
Clearly on chrome there is no gap, so I think this is a browser-dependent behavior. I think we should try to support firefox even though it is not 100% functional at the moment.

File: javascript/compute-height.js (hunk 1 above)
Comment (John-WL) [reply javascript/compute-height.js:18]: and then call setTimeout twice?
```

    if(dynamicElement === null) {
        // polling ew
        setTimeout(computeComfyuiElementHeight, 200); // 1
        return;
    }

    const height = dynamicElement.offsetHeight;
    container.style.height = `calc(100% - ${height-TAB_OFFSET_PADDING}px)`;
    setTimeout(computeComfyuiElementHeight, 200); // 2
```

Comment (John-WL) [reply javascript/compute-height.js:18]: else, we can do this:
```js
    // polling ew
    setTimeout(computeComfyuiElementHeight, 200);

    if(dynamicElement === null) {
        return;
    }

    const height = dynamicElement.offsetHeight;
    container.style.height = `calc(100% - ${height-TAB_OFFSET_PADDING}px)`;
```

but I'd like for the setTimeout to be at the end for clarity

**Comment by John-WL** [comment]
Yes, the footer is still not removed in firefox. Not sure how to do this cleanly with polling in mind. I think we really need an event-based method for updating the height.

**Comment by John-WL** [comment]
can you ask gpt4 for a firefox version of this?
```css

#tabs:has( > div#tab_comfyui_webui_root[style*="display: block;"]) ~ #footer {
    display: none;
}
```

**Comment by ljleb** [comment]
Something like this?
```
window.onload = function() {
    var tabElement = document.querySelector('#tabs > #tab_comfyui_webui_root');
    var footerElement = document.querySelector('#tabs ~ #footer');
    if (window.getComputedStyle(tabElement).display === 'block') {
        footerElement.style.display = 'none';
    }
};
```

Probably shouldn't do this on load, but this could work.

**Comment by ljleb** [comment]
If it's possible to call this function on tab change it would work. You just need to keep track of the last activation state of the comfyui tab and toggle the display property when the tab display changes.

**Comment by John-WL** [comment]
nvm, I have a very dumb css/js solution that works

File: javascript/compute-height.js (hunk 1 above)
Comment (ljleb) [reply javascript/compute-height.js:20]: My issue is with the nesting of the function code, but I guess in this case it's not really possible without duplicating the weird return logic with setTimeout. I guess you can pick your poison.

**Review by ljleb — APPROVED** [review]

**Comment by ljleb** [comment]
LGTM

Please, make sure to consider fixing the top padding for firefox before merging.

**Comment by John-WL** [comment]
> LGTM
> 
> Please, make sure to consider fixing the top padding for firefox before merging.

I tried it on my side, it looks fine as is. Is it still broken for you?

**Comment by ljleb** [comment]
It does work after testing again. There's a slight lag but it shouldn't be a problem. Looks good.

**Pull request merged**